python/tests/avro/test_encoder.py

```diff
@@ -0,0 +1,207 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import datetime
+import io
+import struct
+from decimal import Decimal
+
+from pyiceberg.avro.encoder import BinaryEncoder
+
+
+def test_write() -> None:
+    output = io.BytesIO()
+    encoder = BinaryEncoder(output)
+
+    _input = b"\x12\x34\x56"
+
+    encoder.write(_input)
+
+    assert output.getbuffer() == _input
+
+
+def test_write_boolean() -> None:
+    output = io.BytesIO()
+    encoder = BinaryEncoder(output)
+
+    encoder.write_boolean(True)
+    encoder.write_boolean(False)
+
+    assert output.getbuffer() == struct.pack("??", True, False)
+
+
+def test_write_int() -> None:
```

Comment:
I think this should also have a suite of tests that validates round-trip serialization using encoder and decoder. That would have caught the UUID issue because the decoder is implemented correctly.